File: client-app/src/app/models/higherEducationFacility.ts
import { City, RegionDto } from "./region";

export interface HigherEducationFacility {
	id: string;
	name: string;
	accreditation: Accreditation;
	region: RegionDto;
	city: City;
	address: string;
	rating: number;
	website: string;
	info: string;
	telephone: string;
	ukraineTop: number;
	studentsCount: number;
	titlePhoto: string;
	location: Location;
	inFavoriteList: boolean;
}

export interface Accreditation {
	id: number;
	accreditationLevel: string;
}

export interface Location {
	id: number;
	longitude: number;
	latitude: number;
}

export interface HigherEducationFacilityFormValues {
	id: string;
	name: string;
	accreditation: number;
	region: number;
	city: number;
	address: string;
	website: string;
	info: string;
	telephone: string;
	ukraineTop: number;
	titlePhoto: string;
	locationLat?: number;
	locationLong?: number;
}

export interface HigherEducationFacilityQueryParams {
	name: string,
	accreditationId: number,
	regionsId: number[],
	citiesId: number[],
	degreeId: number,
	knowledgeBranchesId: string[],
	specialtyBasesId: string[],
	minPrice: number,
	maxPrice: number,
	budget: boolean,
	ukraineTop: boolean
}

File: Domain/HigherEducationFacility.cs
namespace Domain
{
	public class HigherEducationFacility
	{
		public Guid Id { get; set; }
		public string Name { get; set; }
		public Region Region { get; set; }
		public City City { get; set; }
		public Location Location { get; set; }
		public Accreditation Accreditation { get; set; }
		public string Address { get; set; }
		public string Website { get; set; }
		public string Info { get; set; }
		public string Telephone { get; set; }
		public int UkraineTop { get; set; }
		public string TitlePhoto { get; set; }
		public ICollection<Photo> Photos { get; set; } = new List<Photo>();
		public ICollection<Faculty> Faculties { get; set; } = new List<Faculty>();
		public ICollection<Review> Reviews { get; set; } = new List<Review>();
		public ICollection<HigherEducationFacilityAdmin> HigherEducationFacilityAdmins { get; set; } = new List<HigherEducationFacilityAdmin>();
		public ICollection<FavoriteList> FavoriteList { get; set; } = new List<FavoriteList>();
	}
}